src/main/java/edu/asu/plp/tool/backend/isa/events/RegWatchResponseEvent.java

```diff
@@ -2,32 +2,74 @@
 
 import javafx.beans.property.LongProperty;
 
+/**
+ * @brief Event to denote Register Watch Result. It returns a reference of the register object.
+ * 	      It offers communication between the front-end and the back-end
+ */
 public class RegWatchResponseEvent {
 
+	/**
+	 * @brief Boolean Variable to store the status of the request result(True or False)
+	 */
 	private boolean success;
+
+	/**
+	 * @brief String variable to hold the registerID
+	 */
 	private String registerID;
+
+	/**
+	 * @brief String variable to hold the registerName
+	 */
 	private String registerName;
+
+	/**
+	 * @brief LongProperty class defines a Property wrapping a long value.
+	 */
 	private LongProperty regObject;
 
+	/**
+	 * @brief This is a constructor for which accepts Success, ErrorMessage, ProjectName and ASMImage parameters
+	 * @param success       - It takes the register watch success parameter
```

Comment:
```
@param success indicates the watch request result; true means succeed, otherwise failed.
@param registerID the ID related to the back-end register object requested by the front-end.
@param registerName the name related to the back-end register object requested by the front-end.
@param regObject if the watch result is true, a valid reference to the register object, otherwise null.  
```